magnet/src/test/java/magnet/inspection/events/OnEnterScope.java

```diff
@@ -0,0 +1,36 @@
+package magnet.inspection.events;
+
+import magnet.Scope;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.Objects;
+
+public class OnEnterScope {
```

Comment:
Nit: I know Magnet should be Java-compatible, but why not at least write these tests using Kotlin? This would avoid so much boilerplate code here as it could be written as a simple `data class`, no?

Reply:
It's a trade off between using java in tests or having two different build configurations in same project. I prefer the first one. In my opinion, generating methods in java classes is not so convenient but less painful that having potential issues with build configuration.